bgpd/bgp_route.c

```diff
@@ -295,6 +296,85 @@ struct bgp_path_info *bgp_path_info_unlock(struct bgp_path_info *path)
 	return path;
 }
 
+/* This function sets flag BGP_NODE_SELECT_DEFER based on condition */
+static int bgp_node_set_defer_flag(struct bgp_node *rn, bool delete)
+{
+	struct peer *peer;
+	struct bgp_path_info *old_pi, *nextpi;
+	bool set_flag = 0;
+	struct bgp *bgp = NULL;
+	struct bgp_table *table = NULL;
+	afi_t afi = 0;
+	safi_t safi = 0;
+	char buf[PREFIX2STR_BUFFER];
+
+	/* If the flag BGP_NODE_SELECT_DEFER is set and new path is added
+	 * then the route selection is deferred
+	 */
+	if (CHECK_FLAG(rn->flags, BGP_NODE_SELECT_DEFER) && (delete == false))
+		return 0;
+
+	if (CHECK_FLAG(rn->flags, BGP_NODE_PROCESS_SCHEDULED)) {
+		if (BGP_DEBUG(update, UPDATE_OUT)) {
+			prefix2str(&rn->p, buf, PREFIX2STR_BUFFER);
+			zlog_debug("Route %s is in workqueue and being processed, not deferred.",
+				buf);
+		}
+		return 0;
+	}
+
+	table = bgp_node_table(rn);
+	if (table) {
+		bgp = table->bgp;
+		afi = table->afi;
+		safi = table->safi;
+	}
```

Comment:
table pointer should exist in all normally cases. However adding a check for table pointer along with set_flag would be better at line no 362, instead of assert.